tests/ottest/fixtures/blockchain/Restart.cpp

```diff
@@ -0,0 +1,264 @@
+    Vector<UTXO>& all_bobs_outputs,
+    Vector<UTXO>& all_alice_outputs,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_bob,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_alice)
+{
+    const auto& bobs_network =
+        user_bob.api_->Network().Blockchain().GetChain(test_chain_);
+    const auto& bobs_wallet = bobs_network.Wallet();
+
+    const auto& alice_network =
+        user_alice.api_->Network().Blockchain().GetChain(test_chain_);
+    const auto& alice_wallet = alice_network.Wallet();
+
+    all_bobs_outputs = bobs_wallet.GetOutputs(TxoState::All);
+    all_alice_outputs = alice_wallet.GetOutputs(TxoState::All);
+
+    auto all_types = {
+        TxoState::Immature,
+        TxoState::ConfirmedSpend,
+        TxoState::UnconfirmedSpend,
+        TxoState::ConfirmedNew,
+        TxoState::UnconfirmedNew,
+        TxoState::OrphanedSpend,
+        TxoState::OrphanedNew};
+    for (const auto& type : all_types) {
+        number_of_outputs_per_type_bob[type] =
+            bobs_wallet.GetOutputs(type).size();
+        number_of_outputs_per_type_alice[type] =
+            alice_wallet.GetOutputs(type).size();
+    }
+}
+
+void Restart_fixture::collectOutputsAsSet(
+    const User& user_bob,
+    const User& user_alice,
+    OutputsSet& all_bobs_outputs,
+    OutputsSet& all_alice_outputs,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_bob,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_alice)
+{
+    // Wait for outputs to process
+    sleep(20);
+
+    Vector<UTXO> bobs_outputs;
+    Vector<UTXO> alice_outputs;
+
+    collectOutputs(
+        user_bob,
+        user_alice,
+        bobs_outputs,
+        alice_outputs,
+        number_of_outputs_per_type_bob,
+        number_of_outputs_per_type_alice);
+
+    EXPECT_EQ(bobs_outputs.size(), bobs_outputs.size());
+    EXPECT_EQ(alice_outputs.size(), alice_outputs.size());
+
+    for (std::size_t i = 0; i < bobs_outputs.size(); ++i)
+        all_bobs_outputs.insert(std::move(bobs_outputs[i]));
+
+    for (std::size_t i = 0; i < alice_outputs.size(); ++i)
+        all_alice_outputs.insert(std::move(alice_outputs[i]));
+}
+
+std::pair<ot::Amount, ot::Amount> Restart_fixture::collectFeeRate(
+    const User& user_bob,
+    const User& user_alice)
+{
+    const auto& bobs_network =
+        user_bob.api_->Network().Blockchain().GetChain(test_chain_);
+
+    const auto& alice_network =
+        user_alice.api_->Network().Blockchain().GetChain(test_chain_);
+
+    return std::make_pair(
+        bobs_network.Internal().FeeRate(), alice_network.Internal().FeeRate());
+}
+
+void Restart_fixture::validateOutputs(
+    const User& user_bob_after_reboot,
+    const User& user_alice_after_reboot,
+    const std::set<UTXO>& bob_outputs,
+    const std::set<UTXO>& alice_outputs,
+    const std::map<TxoState, std::size_t>& bob_all_outputs_size,
+    const std::map<TxoState, std::size_t>& alice_all_outputs_size)
```

Comment:
again, please split it so it is more generic and just call it twice in the test code ;)